File: ProjectFinal/BE/ParkingManagement/DataAccessLayer/Repositories/Interfaces/IUserRepository.cs
using DomainLayer.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Repositories.Interfaces
{
    public interface IUserRepository : IGenericRepository<User>
    {
        Task<User?> GetByEmailAsync(string email);
    }
}

File: VBNet/MyLibrary.DAL/Repositories/Interfaces/IUserRepository.vb
Imports MyLibrary.Domain

Public Interface IUserRepository
    Inherits IGenericRepository(Of User)

    Function GetByEmailAsync(email As String) As Task(Of User)
    Function ExistsByEmailAsync(email As String) As Task(Of Boolean)

    Function GetAllIncludedDeleted() As IQueryable(Of User)
End Interface
